### ISSUE TITLE
FAILED: TcpSpec must not shut down connections after stream cancelled

### ISSUE THREAD

**Issue by ktoso** [akka/akka#20284]
```
[WARN] [04/11/2016 00:03:49.776] [TcpSpec-akka.actor.default-dispatcher-10] [akka://TcpSpec/system/IO-TCP/selectors/$a/62] received dead letter from Actor[akka://TcpSpec/user/StreamSupervisor-170/$$L#-411612188]: ResumeReading
[info] - must not shut down connections after the connection stream cancelled *** FAILED *** (1 second, 29 milliseconds)
[info]   akka.stream.StreamTcpException: The connection closed with error: Connection reset by peer
```

https://jenkins.akka.io:8498/job/akka-multi-node-nightly/3709/consoleFull

**Comment by drewhk** [comment]
https://jenkins.akka.io:8498/job/akka-multi-node-nightly/3736/consoleFull

**Comment by johanandren** [comment]
https://jenkins.akka.io:8498/job/pr-validator-per-commit-jenkins/5950/consoleFull

**Comment by drewhk** [comment]
https://jenkins.akka.io:8498/job/akka-multi-node/1581/consoleFull

**Comment by 2m** [comment]
https://jenkins.akka.io:8498/job/akka-multi-node-nightly/4140/consoleFull

**Comment by johanandren** [comment]
I think I fixed this one as well in #21685

**Comment by johanandren** [comment]
I'll close it based on thinking that.
